pkg/controllers/user.go

```diff
@@ -21,10 +23,51 @@ func NewUserController(
 	return &UserController{logger, createUserInteractor}
 }
 
+func (uc *UserController) writeResponse(w http.ResponseWriter, status int, err string) {
+	js, jsonErr := json.Marshal(entities.StatusResponse{Code: status, Message: err})
+	if jsonErr != nil {
+		uc.logger.Errorf("Unable to marshal json: %s", jsonErr)
+		return
+	}
+
+	w.WriteHeader(status)
+	w.Header().Set("Content-Type", "application/json")
+	_, writeErr := w.Write(js)
+	if writeErr != nil {
+		uc.logger.Errorf("Unable to write to connection HTTP: %s", writeErr)
+		return
+	}
+}
+
 // Create handles the server request and calls CreateUserInteractor
 func (uc *UserController) Create(w http.ResponseWriter, r *http.Request) {
-	// TODO: parse request
+	var err error
+	var errStr string
+	var status int
+	var user entities.User
+	var decoder *json.Decoder
+
 	uc.logger.Debug("Handle request to create user")
 
-	uc.createUserInteractor.Execute()
+	decoder = json.NewDecoder(r.Body)
+
+	err = decoder.Decode(&user)
+	if err != nil {
+		uc.logger.Errorf("Invalid request format: %s", err)
+		status = http.StatusUnprocessableEntity
+		errStr = err.Error()
+		goto done
+	}
+
+	err = uc.createUserInteractor.Execute(user)
+	if err != nil {
+		uc.logger.Errorf("Response error: %s", err)
+		status = http.StatusInternalServerError
+		errStr = err.Error()
+	} else {
```

Comment:
Maybe using `goto` above could remove this `else` clause. As we are adding complexity by using `goto` we can avoid this another complexity addition.